qiskit_nature/mappers/second_quantization/linear_mapper.py

```diff
@@ -12,16 +12,135 @@
 
     def map(self, second_q_op: SpinOp) -> PauliSumOp:
-        # TODO
-        raise NotImplementedError()
+
+        qubit_ops_list: List[PauliSumOp] = []
+
+        # get transformed general spin matrices
+        spinx, spiny, spinz, identity = self._linear_encoding(second_q_op.spin)
+
+        for idx, (_, coeff) in enumerate(second_q_op.to_list()):
+
+            operatorlist: List[PauliSumOp] = []
+
+            for n_x, n_y, n_z in zip(second_q_op.x[idx], second_q_op.y[idx], second_q_op.z[idx]):
+
+                operator_on_spin_i: List[PauliSumOp] = []
+
+                if n_x > 0:
+                    # construct the qubit operator embed
+                    operator_on_spin_i.append(
+                        reduce(operator.matmul, [spinx for i in range(int(n_x))])
+                    )
+
+                if n_y > 0:
+                    # construct the qubit operator embed
+                    operator_on_spin_i.append(
+                        reduce(operator.matmul, [spiny for i in range(int(n_y))])
+                    )
+
+                if n_z > 0:
+                    # construct the qubit operator embed
+                    operator_on_spin_i.append(
+                        reduce(operator.matmul, [spinz for i in range(int(n_z))])
+                    )
+
+                if np.any([n_x, n_y, n_z]) > 0:
+                    # multiply X^n_x * Y^n_y * Z^n_z
+                    single_operator_on_spin_i = reduce(operator.matmul, operator_on_spin_i)
+                    operatorlist.append(single_operator_on_spin_i.reduce())
+
+                else:
+                    # If n_x=n_y=n_z=0, simply add the embedded Identity operator.
+                    operatorlist.append(identity)
+
+            # Now, we can tensor all operators in this list
+            # NOTE: in Qiskit's opflow the `XOR` (i.e. `^`) operator does the tensor product
+            qubit_ops_list.append(coeff * reduce(operator.xor, reversed(operatorlist)))
+
+        qubit_op = reduce(operator.add, qubit_ops_list)
+
+        return qubit_op
+
+    def _linear_encoding(self, spin: Union[Fraction, float]) -> List[PauliSumOp]:
+        """
+        Generates a 'linear_encoding' of the spin S operators 'X', 'Y', 'Z' and 'identity'
+        to qubit operators (linear combinations of pauli strings).
+        In this 'linear_encoding' each individual spin S system is represented via
+        2S+1 qubits and the state |s> is mapped to the state |00...010..00>, where the s-th qubit is
+        in state 1.
+        Returns:
+            self.transformed_XYZI: list,
+                The 4-element list of transformed spin S 'X', 'Y', 'Z' and 'identity' operators.
+                I.e.
+                    self.transformed_XYZI[0] corresponds to the linear
+                    combination of pauli strings needed
+                    to represent the embedded 'X' operator
+        """
+
+        trafo_xzyi: List[PauliSumOp] = []
```

Comment:
Just a comment - strange name - I had wondered if trafo was some math term or persons name for some construct they defined. I am guessing its short for transformed?